src/main/jbake/content/blog/2017/01/13/events-aggregation-extension.adoc

```diff
@@ -0,0 +1,189 @@
+= Extending Complex Event Processing in Hawkular Alerting
+Lucas Ponce
+2017-01-13
+:jbake-type: post
+:jbake-status: published
+:jbake-tags: blog, alerts, complex, events, processing, stream, polling
+
+== Stream Processing versus Polling Processing
+
+Hawkular Alerting uses different techniques to detect behaviours by defining rules.
+
+The main implementation is based in a Stream Processing design.
+Hawkular Alerting analyzes streams of data and events ordered in a timeline searching for matches against conditions.
+This technique is also known as Event Processing (or Complex Event Processing) specially when the processing of incoming data might need multiple conditions.
+
+ifndef::env-github[]
+image::/img/blog/2017/stream-processing.png[Stream Processing]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2017/stream-processing.png[Stream Processing]
+endif::[]
+
+This method is particularity good to identify meaningful happenings on complex scenarios and respond to them as quickly as possible.
+
+Hawkular Alerting also supports a Polling technique based on _Alerters_ that allow us to define queries that are processed
+against a backend in a periodic manner.
```

Comment:
Overall, the Alerters approach is more broad than just query pollers, but for the purpose of this blog I think it is OK to be described this way.  This suggeested restructuring may be better:

Hawkular Alerting also supports an _Alerter_ based Polling technique that allows periodic processing of defined queries against a backend.